first_order_model/train.py

```diff
@@ -186,58 +184,58 @@ def train(config, generator, discriminator, kp_detector, checkpoint, log_dir, da
         optimizer_kp_detector = torch.optim.Adam(generator_full.kp_extractor.parameters(), 
                 lr=train_params['lr_kp_detector'], betas=(0.5, 0.999))
         optimizer_discriminator = torch.optim.Adam(generator_full.discriminator.parameters(), lr=train_params['lr_discriminator'], betas=(0.5, 0.999))
-        current = get_model_macs(log_dir, copy.deepcopy(generator_full.generator), kp_detector, torch.device('cuda' if torch.cuda.is_available() else 'cpu'), lr_size, 512, 2)
+        current = get_decode_and_bottleneck_macs(log_dir, copy.deepcopy(generator_full.generator), kp_detector, torch.device('cuda' if torch.cuda.is_available() else 'cpu'), lr_size, image_shape, 2)
         discriminator_full = DiscriminatorFullModel(generator_full.kp_extractor, generator_full.generator, generator_full.discriminator, train_params)
+        generator = generator_full.generator
+        discriminator = generator_full.discriminator
+        kp_detector = generator_full.kp_extractor
         print('reloaded params, new macs is', current)
 
-    for x in dataloader:
-        break
 
     if train_params.get('netadapt', False):
         sort = train_params.get('netadapt_sort', False)
         steps_per_it = train_params.get('netadapt_steps_per_it', -1)
         with Logger(log_dir=log_dir, visualizer_params=config['visualizer_params'], checkpoint_freq=train_params['checkpoint_freq']) as logger:
-            epoch = 0
-            is_first_round = True
+            netadapt_iteration = 0
+
             while current > target:
-                epoch += 1
                 print("Current macs is: ", current)
                 print("Start macs was: ", start)
                 print("Shrink ratio is: ", current/start)
-                if not is_first_round:
 
+                if netadapt_iteration != 0:
                     # Run one iteration of netadapt
-
                     generator_full.generator, generator_full.kp_extractor, generator_full.discriminator = reduce_macs(
                         generator_full.generator, current - reduce_amount,
                         current, generator_full.kp_extractor, generator_full.discriminator, train_params,
                         dataloader, metrics_dataloader, generator_type,
-                        lr_size, generator_full, sort, steps_per_it, device_ids, log_dir, discriminator_full)
+                        lr_size, generator_full, sort, steps_per_it, device_ids, log_dir, discriminator_full, image_shape)
                     with warnings.catch_warnings():
                         warnings.simplefilter("ignore")
-                        current = get_model_macs(log_dir, generator_full.generator, kp_detector, torch.device('cuda' if torch.cuda.is_available() else 'cpu'), lr_size, 512, 2)
+                        current = get_decode_and_bottleneck_macs(log_dir, generator_full.generator, kp_detector, torch.device('cuda' if torch.cuda.is_available() else 'cpu'), lr_size, image_shape, 2)
```

Comment:
break up long lines. use a variable for device

Reply:
Done.